pkg/operation/operation.go

```diff
@@ -436,7 +436,7 @@ func (o *Operation) EnsureShootStateExists(ctx context.Context) error {
 	blockOwnerDeletion := false
 	ownerReference.BlockOwnerDeletion = &blockOwnerDeletion
 
-	_, err := controllerutil.CreateOrUpdate(ctx, o.K8sGardenClient.Client(), shootState, func() error {
+	_, err := controllerutil.CreateOrUpdate(ctx, o.K8sGardenClient.DirectClient(), shootState, func() error {
```

Comment:
Have you seen any issues here with the cached client?
If yes, I would probably also go for a patch here instead of using the `DirectClient`.

Reply:
Not in this function directly, however we ran into a case where the `Get()` inside `CreateOrUpdate()` would return `NotFound` error when the object actually exists in the cluster.
The first time EnsureShootStateExists is executed it should create the ShootState object. Generally, this means that it should be ok to use the cached client, but after gardenlet restarts we might encounter the error I mentioned above.

Reply:
Hmm, ok. Thanks for explaining.